server.js

```diff
@@ -20,4 +24,80 @@ app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+// returining all messages
+app.get("/messages", (req, res) => {
+  const queryText = req.query.text;
+  messages.forEach((message) => console.log(message.text.includes(queryText)));
+
+  // console.log(messages.filter((message) => message.text.includes(queryText)));
+  if (queryText) {
+    return res.json(
+      messages.filter((message) =>
+        message.text.toLowerCase().includes(queryText.toLowerCase())
+      )
+    );
+  }
+
+  res.json(messages).send;
+});
+
+// return the result of the latest 10 messages
+app.get("/messages/latest", (req, res) => {
+  const latestTenMessages = -10;
+  res.json(messages.slice(latestTenMessages));
```

Comment:
this is good 👍 
think about how you could change this to query for a specific number of messages